plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/preferences/PrefPageSQLCodeEditing.java

```diff
@@ -144,7 +144,8 @@ protected void loadPreferences(DBPPreferenceStore store) {
         csProblemMarkersEnabled.setSelection(store.getBoolean(SQLPreferenceConstants.PROBLEM_MARKERS_ENABLED));
         advancedHighlightingEnabled.setSelection(store.getBoolean(SQLPreferenceConstants.ADVANCED_HIGHLIGHTING_ENABLE));
         readMetadataForSemanticValidationEnabled.setSelection(store.getBoolean(SQLPreferenceConstants.READ_METADATA_FOR_SEMANTIC_ANALYSIS));
-
+        readMetadataForSemanticValidationEnabled.setEnabled(advancedHighlightingEnabled.getSelection());
```

Comment:
So it depends on the `ADVANCED_HIGHLIGHTING_ENABLE`? Why do we have two checkboxes for one option then?

Reply:
We can use metadata for semantic analysis or now. It's a separate checkbox for that, which should be disabled, if checkbox about semantic analysis is turned off. We have this for a long time, I just fixed synchronization on settings loading.

Reply:
I see, it's `setEnabled`